bayesian/observability/metrics.py

```diff
@@ -0,0 +1,83 @@
+"""Basic Metrics Collection logic."""
+
+import time
+
+from flask import current_app, g, request
+from prometheus_client import Counter, Histogram
+from prometheus_client.metrics import MetricWrapperBase
+from prometheus_client.utils import INF
+from typing import Dict, Optional
+import os
+from prometheus_client import CollectorRegistry
+from prometheus_client.multiprocess import MultiProcessCollector
+
+
+METRICS_PARAMS = {
+    "requests": (
+        Counter,
+        {
+            "name": "http_requests_total",
+            "documentation": "Total count of requests by method, path, and status code.",
+            "labelnames": ["method", "path_template", "status_code"],
+        },
+    ),
+    "requests_duration": (
+        Histogram,
+        {
+            "name": "http_requests_duration_seconds",
+            "documentation": "Histogram of requests processing time by path (in seconds)",
+            "labelnames": ["method", "path_template", "status_code_family"],
+            "buckets": (0.1, 0.5, 1, 5, 10, INF),
+        },
+    )
+}
+
+
+def get_metrics_registry() -> CollectorRegistry:
+    """Initialize a observability registry."""
+    prometheus_multiproc_dir = os.environ.get('PROMETHEUS_MULTIPROC_DIR')
+    registry = CollectorRegistry()
+    if prometheus_multiproc_dir:
+        MultiProcessCollector(registry, path=prometheus_multiproc_dir)
+    return registry
+
+
+def init_metrics(registry: CollectorRegistry) -> Dict[str, MetricWrapperBase]:
+    """Initialize the observability with the registry."""
+    metrics = {}
+    for name, init_bits in METRICS_PARAMS.items():
+        metric_type, params = init_bits
+        metrics[name] = metric_type(registry=registry, **params)
+    return metrics
+
+
+def emit_response_metrics(status_code: int, metrics: Dict[str, MetricWrapperBase]) -> None:
```

Comment:
I feel this method will grow as we keep adding metrics. Can this be split for each metric?

Reply:
Ack.